app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb

```diff
@@ -11,7 +11,7 @@ snippet: false
 :net<%= i %>
 isset ${net<%= i -%>/mac} || goto no_nic
 dhcp net<%= i -%> || goto net<%= i+1 %>
-chain <%= foreman_url('iPXE', mac: "${net#{i}/mac}") %> || goto net<%= i+1 %>
+chain <%= foreman_url('iPXE', raw: { mac: "${net#{i}/mac}" }) %> || goto net<%= i+1 %>
```

Comment:
How about changing it directly here instead of modifying the generic url helper and introducing an additional parameter?
```suggestion
chain <%= foreman_url('iPXE')+"?${net#{i}/mac}" %> || goto net<%= i+1 %>
```

Reply:
Another option is to explicitly unescape it here:
```suggestion
chain <%= CGI.unescape(foreman_url('iPXE', mac: "${net#{i}/mac}" )) %> || goto net<%= i+1 %>
```

Reply:
We used to have it like this, but we were running into issues with `?` vs `&` characters. You can only use `?` when it is the first parameter, however `foreman_url` adds a token and it can be also disabled by Administer Setting value. That's why this is handled by the helper, I think it's more robust to handle this logic in there and cover this with unit tests - which I did.

By unescaping you are actually breaking the other parameters, granted we don't have any non-alphanum characters today, this can change.

What exactly is bad with my patch?

Reply:
I was hoping we could solve it in a simpler way that doesn't add complexity to the main `foreman_url` method that is used everywhere. Also, adding the `has_raw=yes` parameter to the url seems odd.

Reply:
Well if you find cleaner solution let me know. I can remove the `has_raw` one in exchange for more ugly code.

Reply:
I think a parameter `unescaped_get` is clearer than `raw`. I'd also like to see a solution that doesn't have `has_raw`. I still think that having the `foreman_url` helper solve it is cleaner than trying to work around it in the template.